File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map();
        for (let i = 0 ; i < nums.length; i++)
        {
            let difference = target - nums[i];
            if (!map.has(difference))
            {
                map.set(nums[i], i);
            }
            else
            {
                return [i,map.get(target-nums[i])].sort()
            }
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.Length - 1;

        while(i<j)
        {
            if(numbers[i] + numbers[j] > target)
                j = j-1;
            if(numbers[i] + numbers[j] < target)
                i = i+1;
            if(numbers[i] + numbers[j] == target)
                break;

        }

        return [i+1,j+1];
    }
}
